backend/src/models/user.js

```diff
@@ -0,0 +1,40 @@
+module.exports = (sequelize, DataTypes) => {
+  const User = sequelize.define('User', {
+    userId: {
+      allowNull: false,
+      defaultValue: DataTypes.UUIDV4,
+      primaryKey: true,
+      type: DataTypes.UUID,
+      validate: {
+        isUUID: 4
+      }
+    },
+    keycloakId: {
+      allowNull: false,
+      comment: 'Associated Keycloak user id',
+      type: DataTypes.UUID,
+      unique: true,
+      validate: {
+        isUUID: 4
+      }
+    },
+    displayName: {
```

Comment:
Should we be saving which system has defined the displayName?  BCEID, IDIR, GitHub?
One thing we should include in the model is which Keycloak instance this id is from...  Perhaps we need another model/table to track that...

Reply:
- I've just added a username field which will be Keycloak's representation of what the user and system is from (ie steve@idir or bob@github).
- I'm not sure why we'd need to include which instance of Keycloak an ID is coming from - there is only ever going to be one Keycloak that the application will authenticate to because it is the source of truth.
- There was some discussion about whether or not we should just consolidate the userId and keycloakId together to be one sole primary key. We could go either way and there are merits for different naming and structure. May need more discussion on that front.